### PR TITLE
2019-10-01 GUARD-225 Broken Inventory Sync

### PR DESCRIPTION
**Pull request by fabulaspb**
Summary

Our customer which has listings with a lot of variations (100 and more per listing) faced API limitations during inventory update:
`You have exceeded your maximum call limit of 250 for item per day. Try back after 1 day`

To reduce the number of API calls I grouped all request objects which have the same itemId into the one ( one item can have several variations).

### CODE REVIEW

File: src/EbayAccess/Models/ReviseFixedPriceItemRequest/ReviseFixedPriceItemRequest.cs [hunk 1]
```diff
@@ -1,12 +1,59 @@
...
+		{
+			var result = new ReviseFixedPriceItemRequest() { ItemId = source.Key };
+
+			if ( source.Count() > 1 )
+			{
+				result.Variations.AddRange( source.Select( i => new ReviseFixedPriceItemVariationRequest() { Quantity = i.Quantity, Sku = i.Sku, StartPrice = i.StartPrice } ) );
+			}
+			else
+			{
+				result.Sku = source.First().Sku;
```
Comment (maxim-skuvault) [inline src/EbayAccess/Models/ReviseFixedPriceItemRequest/ReviseFixedPriceItemRequest.cs]: Is it possible for source.Count() to be 0? If so, then these will blow up

File: src/EbayAccessTests.Integration/TestEnvironment/TestBase.cs [hunk 2]
```diff
@@ -29,8 +29,8 @@ public void Init()
 
 	internal static class ExistsProductsCreatedInRange
 	{
-		public static readonly DateTime DateFrom = new DateTime( 2014, 6, 14, 15, 0, 0, DateTimeKind.Local );
-		public static readonly DateTime DateTo = new DateTime( 2014, 9, 24, 15, 0, 0, DateTimeKind.Local );
+		public static readonly DateTime DateFrom = new DateTime( 2019, 10, 1, 15, 0, 0, DateTimeKind.Local );
```
Comment (maxim-skuvault) [inline src/EbayAccessTests.Integration/TestEnvironment/TestBase.cs]: I don't think you can query eBay for GetSellerList with a date range greater than 120 days

File: src/EbayAccess/Models/ReviseFixedPriceItemRequest/ReviseFixedPriceItemRequest.cs (hunk 1 above)
Comment (fabulaspb) [reply src/EbayAccess/Models/ReviseFixedPriceItemRequest/ReviseFixedPriceItemRequest.cs]: It's not possible, source will have at least one element.

File: src/EbayAccessTests.Integration/TestEnvironment/TestBase.cs (hunk 2 above)
Comment (fabulaspb) [reply src/EbayAccessTests.Integration/TestEnvironment/TestBase.cs]: Fixed it.

File: .build.cmd [hunk 3]
```diff
@@ -1,14 +1,11 @@
...
 
-setlocal
-set PATH=%PATH%;tools\Invoke-Build
-
 set LOGDIR=%~dp0log\
 set LOGFILE=%LOGDIR%%1.log
 if not exist %LOGDIR% md %LOGDIR%
 
-powershell -NoProfile -ExecutionPolicy unrestricted "Build -Parameters @{Configuration='%1'} -Summary:$True" -verbose
+tools\Invoke-Build\ib -File:.build.ps1 -Summary -Verbose
```
Comment (maxim-skuvault) [inline .build.cmd]: Are the changes to this file intentional? Why? Are other integrations like that?

Comment (fabulaspb) [reply .build.cmd]: Had issues with running release.cmd. So I copied script body from Magento integration.

File: src/Global/GlobalAssemblyInfo.cs [hunk 4]
```diff
@@ -24,4 +24,4 @@
 
 // Keep in track with CA API version
 
-[ assembly : AssemblyVersion( "1.1.39.0" ) ]
\ No newline at end of file
+[ assembly : AssemblyVersion( "1.1.40.0" ) ]
```
Comment (maxim-skuvault) [inline src/Global/GlobalAssemblyInfo.cs]: If you make changes, then publish as 1.1.42. I published my PR as 1.1.41

**Pull request merged**